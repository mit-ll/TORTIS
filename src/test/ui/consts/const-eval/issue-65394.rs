// Test for absence of validation mismatch ICE in #65394

const _: Vec<i32> = {
    let mut x = Vec::<i32>::new();
    let r = &mut x; //~ ERROR references in constants may only refer to immutable values
    let y = x;
    y
};

fn main() {}
