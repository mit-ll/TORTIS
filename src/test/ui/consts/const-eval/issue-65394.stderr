error[E0017]: references in constants may only refer to immutable values
  --> $DIR/issue-65394.rs:5:13
   |
LL |     let r = &mut x;
   |             ^^^^^^ constants require immutable values

[ERROR rustc_mir::transform::qualify_consts] old validator: [($DIR/issue-65394.rs:5:13: 5:19, "MutBorrow(Mut { allow_two_phase_borrow: false })")]
[ERROR rustc_mir::transform::qualify_consts] new validator: [($DIR/issue-65394.rs:5:13: 5:19, "MutBorrow(Mut { allow_two_phase_borrow: false })"), ($DIR/issue-65394.rs:4:9: 4:14, "LiveDrop")]
error: aborting due to previous error

For more information about this error, try `rustc --explain E0017`.
